File: ProyectosC#/class/Reserva.cs
using System;

public class Reserva
{
    private int Id { get; set; }
    private Cubiculo Cubiculo { get; set; }
    private Usuario Usuario { get; set; }
    private DateTime HoraReserva { get; set; }

    public Reserva(int id, Cubiculo cubiculo, Usuario usuario)
    {
        Id = id;
        Cubiculo = cubiculo;
        Usuario = usuario;
        HoraReserva = DateTime.Now;
    }

    public void Reservar()
    {
        Cubiculo.Reservar(Usuario as Funcionario);
    }

    public void ElimReserva()
    {
        Cubiculo.Liberar();
        Console.WriteLine("Reserva eliminada.");
    }
}

File: Reserva.java

package parking;


public class Reserva{
    private int id;
    private Cubiculos cube;
    private Funcionario funcer;
    
    public Reserva(int id,Cubiculos cube, Funcionario funcer){
        this.id = id;
        this.cube = cube;
        this.funcer = funcer;
    }
    
    public void Reservar(){
        cube.Reservar(funcer);
    }
    
    public void ElimReserva(){
        cube.Liberar(funcer);
    }
    
}
